packages/react-server-dom-webpack/src/__tests__/ReactFlightDOMReply-test.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+// Polyfills for test environment
+global.ReadableStream =
+  require('web-streams-polyfill/ponyfill/es6').ReadableStream;
+global.TextEncoder = require('util').TextEncoder;
+global.TextDecoder = require('util').TextDecoder;
+
+// let serverExports;
+let webpackServerMap;
+let ReactServerDOMServer;
+let ReactServerDOMClient;
+
+describe('ReactFlightDOMReply', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    const WebpackMock = require('./utils/WebpackMock');
+    // serverExports = WebpackMock.serverExports;
+    webpackServerMap = WebpackMock.webpackServerMap;
+    ReactServerDOMServer = require('react-server-dom-webpack/server.browser');
+    ReactServerDOMClient = require('react-server-dom-webpack/client');
+  });
+
+  it('can pass undefined as a reply', async () => {
+    const body = await ReactServerDOMClient.encodeReply(undefined);
+    const missing = await ReactServerDOMServer.decodeReply(
+      body,
+      webpackServerMap,
+    );
+    expect(missing).toBe(undefined);
+
+    const body2 = await ReactServerDOMClient.encodeReply({
+      array: [undefined, null, undefined],
+      prop: undefined,
+    });
+    const object = await ReactServerDOMServer.decodeReply(
+      body2,
+      webpackServerMap,
+    );
+    expect(object.array.length).toBe(3);
+    expect(object.array[0]).toBe(undefined);
+    expect(object.array[1]).toBe(null);
+    expect(object.array[3]).toBe(undefined);
+    expect(object.prop).toBe(undefined);
+    // These should really be true but our deserialization doesn't currently deal with it.
```

Comment:
@eps1lon Could be good to test for this in the other direction too since it's not working as intended I think.